### ISSUE TITLE
Mining Equipment Vendor

### ISSUE THREAD

**Issue by chairbender** [unitystation/unitystation#2304]
# Description
Depends on #2257 so you can actually get mining credits.
Depends on #2303 so you can actually put arbitrary stuff in it and not just prefabs.

Allow miners to get fun stuff for their hard work.

How it should work:
  * Create an SO holding the list of Spawnables you can buy from it, with a price value for each
  * You need to insert a card in order to buy stuff from it, the inserted card's mining credits will be used to make the purchases
  * You can reference the existing vending machine code for this, but keep in mind that this particular machine has infinite stores and don't need a resupply.

Can see it briefly in action here: https://youtu.be/KOtfRZt6F3s?t=1255

Add any items that we've already built to the vendor that should belong in it.

**Comment by PerfectTangent** [comment]
closing in favour of #6151
